File: eZvso/Dialog_CircleArray.vb
Imports System.Windows.Forms
Public Class Dialog_CircleArray
    Private Num As UShort = 4
    Private Angle As Double = 360
    Private blnCenter As Boolean = True
    Private blnPreserveDirection As Boolean = False


    Public Shadows Function ShowDialog(Optional ByRef Num As UShort = 4, _
                                     Optional ByRef Angle As Double = 360, _
                                     Optional ByRef blnCenter As Boolean = True, _
                                     Optional ByRef blnPreserveDirection As Boolean = False) As DialogResult
        Dim res As DialogResult = MyBase.ShowDialog()
        If res = Windows.Forms.DialogResult.OK Then
            With Me
                Num = .Num
                Angle = .Angle
                blnCenter = .blnCenter
                blnPreserveDirection = .blnPreserveDirection
            End With
        End If
        Return res

    End Function

#Region "   ---   事件处理"
    Private Sub txtAngle_TextChanged(sender As Object, e As EventArgs) Handles txtAngle.TextChanged
        Dim txt As TextBox = DirectCast(sender, TextBox)
        Double.TryParse(txt.Text, Angle)
    End Sub
    Private Sub txtNum_TextChanged(sender As Object, e As EventArgs) Handles txtNum.TextChanged
        Dim txt As TextBox = DirectCast(sender, TextBox)
        UShort.TryParse(txt.Text, Num)
    End Sub

    Private Sub CheckBox_preserveDirection_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox_preserveDirection.CheckedChanged
        Dim box As CheckBox = DirectCast(sender, CheckBox)
        If box.Checked Then
            Me.blnPreserveDirection = True
        Else
            Me.blnPreserveDirection = False
        End If
    End Sub
    Private Sub RadioButton_Center_CheckedChanged(sender As Object, e As EventArgs) Handles RadioButton_Center.CheckedChanged, RadioButton_Border.CheckedChanged
        If Me.RadioButton_Center.Checked Then
            Me.blnCenter = True
        Else
            Me.blnCenter = False
        End If

    End Sub

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
        Me.DialogResult = Windows.Forms.DialogResult.OK
        Me.Close()
    End Sub

#End Region

End Class

File: eZvso/Dialog_CircleArray.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using Office = Microsoft.Office.Core;
using Microsoft.VisualBasic;
using System.Xml.Linq;
using Visio = Microsoft.Office.Interop.Visio;
using System.Text;
using System.Linq;
// End of VB project level imports

using System.Windows.Forms;

namespace eZvso
{
    public partial class Dialog_CircleArray
    {
        public Dialog_CircleArray()
        {
            InitializeComponent();
        }
        private UInt16 Num = 4;
        private double Angle = 360;
        private bool blnCenter = true;
        private bool blnPreserveDirection = false;


        public new DialogResult ShowDialog(UInt16 Num = 4, double Angle = 360, bool blnCenter = true, bool blnPreserveDirection = true)
        {
            DialogResult res = base.ShowDialog();
            if (res == DialogResult.OK)
            {
                Dialog_CircleArray with_1 = this;
                Num = with_1.Num;
                Angle = with_1.Angle;
                blnCenter = with_1.blnCenter;
                blnPreserveDirection = with_1.blnPreserveDirection;
            }
            return res;

        }

        #region    ---   事件处理
        public void txtAngle_TextChanged(object sender, EventArgs e)
        {
            TextBox txt = (TextBox)sender;
            double.TryParse(txt.Text, out Angle);
        }
        public void txtNum_TextChanged(object sender, EventArgs e)
        {
            TextBox txt = (TextBox)sender;
            UInt16.TryParse(txt.Text, out Num);
        }

        public void CheckBox_preserveDirection_CheckedChanged(object sender, EventArgs e)
        {
            CheckBox box = (CheckBox)sender;
            if (box.Checked)
            {
                this.blnPreserveDirection = true;
            }
            else
            {
                this.blnPreserveDirection = false;
            }
        }
        public void RadioButton_Center_CheckedChanged(object sender, EventArgs e)
        {
            if (this.RadioButton_Center.Checked)
            {
                this.blnCenter = true;
            }
            else
            {
                this.blnCenter = false;
            }

        }

        public void btnOK_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.OK;
            this.Close();
        }

        #endregion

    }
}
